system/include/emscripten/em_js.h

```diff
@@ -56,7 +56,7 @@
 
 #define EM_JS(ret, name, params, ...)          \
   _EM_JS_CPP_BEGIN                             \
-  extern ret name params;                      \
+  extern ret name params EM_IMPORT(name);      \
```

Comment:
what is this change?

Reply:
Still curious about this?

Reply:
So this macro is used for defining em_js functions.   At link time these are always expected to be undefined since they are implemented in JS.   Unlike the JS library functions  we can't calculate the list of all known EM_JS symbols ahead of time so we use this method of marking them as external.

In the long run this method is preferable, and ideally all JS library functions would be marked like this in the source code.  However that is much larger and more ambitious change.